### PR TITLE
feat(calling): add rtcstat functionality to calling

### PR DESCRIPTION
**Pull request by rarajes2**
# COMPLETES #< [SPARK-529191](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-529191) >

## This pull request addresses

Adds the rtcstats to the calling SDK

## by making the following changes

Able to send the rtcstats with the calling SDK APIs.

Screenshot 2024-09-13 at 7 08 17 PM

**Note**: For HAR file, please check the JIRA ticket.

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

Tested the SDK by linking with the internal-media-core [PR](https://sqbu-github.cisco.com/WebExSquared/webrtc-media-core/pull/265) changes. We can see in the screenshot and HAR file that the stats are being sent.

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/@webex/internal-plugin-metrics/src/rtcMetrics/index.ts [hunk 1]
```diff
@@ -36,17 +35,23 @@ export default class RtcMetrics {
...
    * Initialize the interval.
    *
    * @param {object} webex - The main `webex` object.
-   * @param {string} meetingId - The meeting id.
    * @param {string} correlationId - The correlation id.
+   * @param {string} meetingId - The meeting id (optional).
+   * @param {string} callId - The meeting id.
    */
-  constructor(webex, meetingId, correlationId) {
+  constructor(webex, correlationId, meetingId, callId) {
```
Comment (mkesavan13) [inline packages/@webex/internal-plugin-metrics/src/rtcMetrics/index.ts:50]: Why are we rearranging the params here?

File: packages/@webex/plugin-meetings/src/meeting/index.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ import {
   ClientEventLeaveReason,
   CallDiagnosticUtils,
   CALL_DIAGNOSTIC_CONFIG,
+  RtcMetrics,
```
Comment (mkesavan13) [inline packages/@webex/plugin-meetings/src/meeting/index.ts:13]: The param change [here](https://github.com/webex/webex-js-sdk/blob/b6fd626956725c6047eeca9827724eb847cdc911/packages/%40webex/internal-plugin-metrics/src/rtcMetrics/index.ts#L50) is not handled in this file [here](https://github.com/webex/webex-js-sdk/blob/b6fd626956725c6047eeca9827724eb847cdc911/packages/%40webex/plugin-meetings/src/meeting/index.ts#L6315)

File: packages/calling/src/CallingClient/calling/call.test.ts [hunk 3]
```diff
@@ -346,7 +346,10 @@ describe('Call Tests', () => {
     expect(mockInternalMediaCoreModule.RoapMediaConnection).toBeCalledOnceWith(
       roapMediaConnectionConfig,
       roapMediaConnectionOptions,
-      expect.any(String)
+      expect.any(String),
+      expect.any(Function),
+      expect.any(Function),
+      expect.any(Function)
     );
```
Comment (mkesavan13) [inline packages/calling/src/CallingClient/calling/call.test.ts:353]: What are these 3 functions? Where does it come from?

File: packages/calling/src/CallingClient/calling/call.ts [hunk 4]
```diff
@@ -2381,7 +2415,7 @@ export class Call extends Eventing<CallEventTypes> implements ICall {
    * Setup a listener for roap events emitted by the media sdk.
    */
   private mediaRoapEventsListener() {
-    this.mediaConnection.on(
+    this.mediaConnection?.on(
```
Comment (mkesavan13) [inline packages/calling/src/CallingClient/calling/call.ts:2418]: Why do we need these operators here? Is there a scenario where the `mediaConnection` would not exist now?

**Review by mkesavan13 — CHANGES_REQUESTED** [review]

**Comment by rarajes2** [comment]
Closing this PR. Will raise 2 PRs for this
1. Moving the RtcMetrics to internal-plugin-metrics
2. Introducing RtcStats in the Calling

**Pull request closed without merging**